numba/cpython/setobj.py

```diff
@@ -946,12 +944,13 @@ def _resize(self, payload, nentries, errmsg):
             context.call_conv.return_user_exc(builder, MemoryError,
                                               (errmsg,))
 
-        # Re-insert old entries
+        # Re-insert old entries, decrefing each one due to `_iterate`'s increfs
```

Comment:
I might add, that `_add_key` adds an incref, so we remove the extra incref from `_iterate`